mobsf/MobSF/views/home.py

```diff
@@ -318,13 +319,20 @@ def recent_scans(request, page_size=10, page_number=1):
 @login_required
 @permission_required(Permissions.SCAN)
 def download_apk(request):
-    """Download and APK by package name."""
+    """Download an APK by package name."""
     package = request.POST['package']
     # Package validated in apk_download()
     context = {
         'status': 'failed',
         'description': 'Unable to download APK',
     }
+    if not is_internet_available():
```

Comment:
Please move this logic to `apk_download()` inside `apk_downloader.py` to avoid additional HttpResponse logic in this controller.

Reply:
Let me know if this implemention is fine or an error should be raised when the condition is true.